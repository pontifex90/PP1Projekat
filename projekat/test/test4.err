ERROR 2017-02-11 06:18:55,390 [main][MJParser.java:778] : Syntax error "=" on line 5, column 14
ERROR 2017-02-11 06:18:55,395 [main][MJParser.java:778] : Semantic error on line 9: non-void method must have return.
ERROR 2017-02-11 06:18:55,395 [main][MJParser.java:778] : Syntax error "aa" on line 13, column 18
ERROR 2017-02-11 06:18:55,397 [main][MJParser.java:778] : Semantic error on line 19: identifier with name "a" already exists.
ERROR 2017-02-11 06:18:55,397 [main][MJParser.java:778] : Semantic error on line 21: tried to use a primitive type as a class object "a"
ERROR 2017-02-11 06:18:55,398 [main][MJParser.java:778] : Syntax error "+" on line 21, column 19
ERROR 2017-02-11 06:18:55,399 [main][MJParser.java:625] : All error messages follow:
				Syntax error "=" on line 5, column 14
				Semantic error on line 9: non-void method must have return.
				Syntax error "aa" on line 13, column 18
				Semantic error on line 19: identifier with name "a" already exists.
				Semantic error on line 21: tried to use a primitive type as a class object "a"
				Syntax error "+" on line 21, column 19
ERROR 2017-02-11 06:18:55,399 [main][MJParser.java:626] : Parsing finished with ERRORS!
