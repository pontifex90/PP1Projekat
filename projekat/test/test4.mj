
program test4

class A {
	int classVar = 3;
	{
		int method(int a) {
		
		}
	}
}

class B extends A aa + {

}

{
	void main(int a[])
	A a;
	{
		a.method(a + a + + =);
	}
}

