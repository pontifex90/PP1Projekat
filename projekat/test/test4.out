INFO  2017-02-11 06:18:55,373 [main][MJParser.java:615] : Compiling source file: D:\PP1ProjekatRepo\PP1Projekat\projekat\test\test4.mj
INFO  2017-02-11 06:18:55,386 [main][MJParser.java:588] : #2 program
INFO  2017-02-11 06:18:55,386 [main][MJParser.java:588] : #49 test4
INFO  2017-02-11 06:18:55,386 [main][MJParser.java:588] : #3 class
DEBUG 2017-02-11 06:18:55,388 [main][MJParser.java:770] : Semantic info on line 2: added program "test4" to symbol table. [ kind: 6  name: test4  type: 0]
INFO  2017-02-11 06:18:55,390 [main][MJParser.java:588] : #49 A
INFO  2017-02-11 06:18:55,390 [main][MJParser.java:588] : #7 {
DEBUG 2017-02-11 06:18:55,390 [main][MJParser.java:770] : Semantic info on line 4: added class type "A" to symbol table. [ kind: 2  name: A  type: 4]
INFO  2017-02-11 06:18:55,390 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:55,390 [main][MJParser.java:588] : #49 classVar
INFO  2017-02-11 06:18:55,390 [main][MJParser.java:588] : #6 =
INFO  2017-02-11 06:18:55,391 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:18:55,391 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:55,391 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:55,391 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:55,392 [main][MJParser.java:588] : #49 method
DEBUG 2017-02-11 06:18:55,393 [main][MJParser.java:874] : Error recovery successful: class variable declaration error (3)
INFO  2017-02-11 06:18:55,393 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:18:55,393 [main][MJParser.java:770] : Semantic info on line 7: added  method "method" to symbol table. [ kind: 3  name: method  type: 1]
INFO  2017-02-11 06:18:55,393 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:55,393 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,393 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:18:55,393 [main][MJParser.java:770] : Semantic info on line 7: added formal parameter variable "a" to symbol table. [ kind: 1  name: a  type: 1]
INFO  2017-02-11 06:18:55,393 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:55,394 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:55,394 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #3 class
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #49 B
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #15 extends
DEBUG 2017-02-11 06:18:55,395 [main][MJParser.java:770] : Semantic info on line 13: added class type "B" to symbol table. [ kind: 2  name: B  type: 4]
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #49 A
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #49 aa
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #28 +
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:55,395 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #17 void
DEBUG 2017-02-11 06:18:55,396 [main][MJParser.java:874] : Error recovery successful: Extends clause error (1)
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #49 main
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:18:55,396 [main][MJParser.java:770] : Semantic info on line 18: added  method "main" to symbol table. [ kind: 3  name: main  type: 0]
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #13 [
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:18:55,396 [main][MJParser.java:770] : Semantic info on line 18: added formal parameter array variable "a" to symbol table. [ kind: 1  name: a  type: 3]
INFO  2017-02-11 06:18:55,396 [main][MJParser.java:588] : #49 A
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #18 .
DEBUG 2017-02-11 06:18:55,397 [main][MJParser.java:770] : Usage info on line 21: formal parameter variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #49 method
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #9 (
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,397 [main][MJParser.java:588] : #28 +
DEBUG 2017-02-11 06:18:55,397 [main][MJParser.java:770] : Usage info on line 21: formal parameter variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #28 +
DEBUG 2017-02-11 06:18:55,398 [main][MJParser.java:770] : Usage info on line 21: formal parameter variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #28 +
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #6 =
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:55,398 [main][MJParser.java:588] : #8 }
DEBUG 2017-02-11 06:18:55,398 [main][MJParser.java:874] : Error recovery successful: Parameters list error (1)
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog test4: notype, 0, 1 
   Type A: Class [Var c0mp@tib13_c11@s_A: notype, 0, 1 Meth method: int, 0, 1 
      Var a: int, 0, 1 
], 0, 0 
   Type B: Class [Var c0mp@tib13_c11@s_B: notype, 0, 1 Var c0mp@tib13_c11@s_noObj: notype, 1, 1 ], 0, 0 
   Meth main: notype, 5, 1 
      Var a: Arr of int, 0, 1 



INFO  2017-02-11 06:18:55,399 [main][MJParser.java:636] : 
===============================
VARS COUNTING:
INFO  2017-02-11 06:18:55,399 [main][MJParser.java:637] : Constants number: 0
INFO  2017-02-11 06:18:55,399 [main][MJParser.java:638] : Global vars number: 0
INFO  2017-02-11 06:18:55,399 [main][MJParser.java:639] : Global array vars number: 0
INFO  2017-02-11 06:18:55,399 [main][MJParser.java:640] : Class vars number: 0
INFO  2017-02-11 06:18:55,399 [main][MJParser.java:641] : Main function vars number: 1
===============================

INFO  2017-02-11 06:18:55,399 [main][MJParser.java:642] : 
===============================
FUNCTION CALLS COUNTING:
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:643] : Global functions: 1
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:644] : Class functions: 1
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:645] : Class static functions: 0
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:646] : Statements block number: 0
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:647] : Main method function calls number: 1
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:648] : Function formal parameters number: 2
INFO  2017-02-11 06:18:55,400 [main][MJParser.java:649] : Class definitions number: 2
